packages/manager/src/features/Billing/BillingPanels/BillingActivityPanel/BillingActivityPanel.tsx

```diff
@@ -302,6 +310,63 @@ export const BillingActivityPanel = React.memo((props: Props) => {
     );
   }, [selectedTransactionType, combinedData]);
 
+  const data =
+    selectedTransactionType.value === 'all' ? combinedData : filteredData;
+
+  const orderedPaginatedData = React.useMemo(() => {
+    const orderedData = data.sort((a, b) => {
+      if (orderBy === 'total') {
+        return order === 'asc' ? a.total - b.total : b.total - a.total;
+      }
+      // If no valid 'orderBy' is provided, return the data as is.
+      return 0;
+    });
+
+    const start = (pagination.page - 1) * pagination.pageSize;
+    const end = start + pagination.pageSize;
+
+    return orderedData.slice(start, end);
+  }, [data, orderBy, order, pagination.page, pagination.pageSize]);
+
+  const renderTableContent = () => {
+    if (accountPaymentsLoading || accountInvoicesLoading) {
+      return <TableRowLoading columns={4} rows={1} />;
```

Comment:
```suggestion
      return <TableRowLoading columns={NUM_COLS} rows={1} />;
```
Can we make the column number a const and use it throughout?